pkg/test-infra/tidb/operation.go

```diff
@@ -108,23 +111,36 @@ func (o *Ops) GetNodes() ([]cluster.Node, error) {
 func (o *Ops) GetClientNodes() ([]cluster.ClientNode, error) {
 	var clientNodes []cluster.ClientNode
 
-	ips, err := util.GetNodeIPs(o.cli, o.ns, map[string]string{"app.kubernetes.io/instance": o.name})
-	if err != nil {
-		return clientNodes, err
-	} else if len(ips) == 0 {
-		return clientNodes, errors.New("k8s node not found")
-	}
-
-	svc, err := o.getTiDBServiceByMeta(&o.tc.Service.ObjectMeta)
-	if err != nil {
-		return clientNodes, err
+	if util.IsInK8sPodEnvironment() {
+		svc, err := o.getTiDBServiceByClusterName(o.ns, o.name)
+		if err != nil {
+			return nil, err
+		}
+		clientNodes = append(clientNodes, cluster.ClientNode{
+			Namespace:   o.ns,
+			ClusterName: o.name,
+			IP:          svc.Spec.ClusterIP,
+			Port:        getTiDBServicePort(svc),
+		})
+	} else {
+		// If case isn't running on k8s pod, uses nodeIP:tidb_port as clientNode for conveniently debug on local
+		ips, err := util.GetNodeIPs(o.cli, o.ns, map[string]string{"app.kubernetes.io/instance": o.name})
```

Comment:
It seems that this will not work on our new K8s since the name looks like `idc-5-46`.

Reply:
I rename it to `GetNodeIPsFromPod(cli client.Client, namespace string, podLabels map[string]string)`, the 3rd parameter is the pod labels.